infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/SenderBufferManager.java

```diff
@@ -234,11 +247,18 @@ public LogReplicationEntry resend() {
             if (entry.timeout(msgTimer) || force) {
                 entry.retry();
                 // Update metadata as topologyConfigId could have changed in between resend cycles
-                LogReplicationEntry dataEntry = entry.getData();
-                dataEntry.getMetadata().setTopologyConfigId(topologyConfigId);
-                CompletableFuture<LogReplicationEntry> cf = dataSender.send(entry.getData());
+                LogReplicationEntryMsg dataEntry = entry.getData();
+                LogReplicationEntryMetadata metadata = LogReplicationEntryMetadata.newBuilder()
+                        .mergeFrom(dataEntry.getMetadata())
+                        .setSiteConfigID(topologyConfigId)
+                        .build();
```

Comment:
same here

Reply:
Same reasoning applies: https://github.com/CorfuDB/CorfuDB/pull/2858#discussion_r568276653

Reply:
Actually, I moved it. Done.